src/ol/sphere/sphere.js

```diff
@@ -58,6 +58,34 @@ ol.Sphere.prototype.cosineDistance = function(c1, c2) {
 
 
 /**
+ * Returns the area using Vincenty for a list of coordinates.
+ *
+ * [Reference](http://trs-new.jpl.nasa.gov/dspace/handle/2014/40409)
+ * Robert. G. Chamberlain and William H. Duquette, "Some Algorithms for
+ * Polygons on a Sphere", JPL Publication 07-03, Jet Propulsion
+ * Laboratory, Pasadena, CA, June 2007
+ *
+ * @param {Array.<ol.Coordinate>} coordinates List of coordinates.
+ * @return {number} Area.
+ * @api
+ */
+ol.Sphere.prototype.vincentyArea = function(coordinates) {
```

Comment:
Looking at this more closely, I don't think this has to do with Vincenty's formulae, because those work on an ellipsoid and not on a sphere. The name you had originally, `geodesicArea`, was fine.